File: spring-boot/jwt/src/main/java/com/example/jwt/auth/ui/AuthController.java
package com.example.jwt.auth.ui;

import com.example.jwt._common.application.dto.ApiResponse;
import com.example.jwt.auth.application.AuthService;
import com.example.jwt.auth.application.dto.RefreshTokenRequest;
import com.example.jwt.auth.application.dto.SignInRequest;
import com.example.jwt.auth.application.dto.SignUpRequest;
import com.example.jwt.auth.application.dto.TokenResponse;
import com.example.jwt.user.application.dto.UserResponse;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {
    private final AuthService authService;

    @PostMapping("/sign-in")
    public ResponseEntity<ApiResponse<TokenResponse>> signIn(
            @RequestBody @Valid SignInRequest signInRequest
    ) {
        TokenResponse tokenResponse = authService.signIn(signInRequest);
        return ResponseEntity
                .status(HttpStatus.OK)
                .body(ApiResponse.success(tokenResponse));
    }

    @GetMapping("/sign-out")
    public ResponseEntity<Void> signOut(
            Authentication auth
    ) {
        authService.signOut(auth);

        return ResponseEntity
                .status(HttpStatus.OK)
                .build();
    }

    @PostMapping("/sign-up")
    public ResponseEntity<Void> signUp(
            @RequestBody @Valid SignUpRequest signUpRequest
    ) {
        authService.signUp(signUpRequest);

        return ResponseEntity
                .status(HttpStatus.CREATED)
                .build();
    }

    @PostMapping("/refresh")
    public ResponseEntity<ApiResponse<TokenResponse>> refresh(
            @RequestBody @Valid RefreshTokenRequest refreshTokenRequest
    ) {
        TokenResponse tokenResponse = authService.refresh(refreshTokenRequest);

        return ResponseEntity
                .status(HttpStatus.OK)
                .body(ApiResponse.success(tokenResponse));
    }

    @GetMapping("/me")
    public ResponseEntity<ApiResponse<UserResponse>> me(
            Authentication auth
    ) {
        UserResponse user = authService.me(auth);

        return ResponseEntity
                .status(HttpStatus.OK)
                .body(ApiResponse.success(user));
    }
}

File: task/payhere/src/main/kotlin/com/example/payhere/domain/auth/controller/AuthController.kt
package com.example.payhere.domain.auth.controller

import com.example.payhere.common.response.EmptyResult
import com.example.payhere.common.response.ResponseCode
import com.example.payhere.common.response.ResponseDto
import com.example.payhere.common.response.SingleResult
import com.example.payhere.domain.auth.controller.dto.SignInRequest
import com.example.payhere.domain.auth.controller.dto.SignInResponse
import com.example.payhere.domain.auth.controller.dto.SignUpRequest
import com.example.payhere.domain.auth.service.AuthService
import com.example.payhere.domain.auth.support.CurrentUser
import com.example.payhere.domain.user.controller.dto.UserDto
import com.example.payhere.domain.user.entity.User
import jakarta.validation.Valid
import org.springframework.http.ResponseEntity
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/auth")
class AuthController(
    private val authService: AuthService
) {
    @PostMapping("/sign-in")
    fun signIn(
        @RequestBody @Valid signInRequest: SignInRequest
    ): ResponseEntity<SingleResult<SignInResponse>> {
        val dto = authService.signIn(signInRequest)
        return ResponseDto.of(ResponseCode.SUCCESS, dto)
    }

    @GetMapping("/sign-out")
    fun signOut(auth: Authentication): ResponseEntity<EmptyResult> {
        authService.signOut(auth.credentials.toString())
        return ResponseDto.of(ResponseCode.SUCCESS)
    }

    @PostMapping("/sign-up")
    fun signUp(
        @RequestBody @Valid signUpRequest: SignUpRequest
    ): ResponseEntity<EmptyResult> {
        authService.signUp(signUpRequest)
        return ResponseDto.of(ResponseCode.SUCCESS)
    }

    @GetMapping("/me")
    fun me(
        @CurrentUser user: User
    ): ResponseEntity<SingleResult<UserDto>> {
        val dto = authService.me(user)
        return ResponseDto.of(ResponseCode.SUCCESS, dto)
    }
}